packages/yarnpkg-pnp/sources/esm-loader/loaderFlags.ts

```diff
@@ -20,3 +20,6 @@ export const SUPPORTS_IMPORT_ATTRIBUTES = major >= 21 || (major === 20 && minor
 
 // https://github.com/nodejs/node/pull/52104
 export const SUPPORTS_IMPORT_ATTRIBUTES_ONLY = major >= 22;
+
+// https://github.com/nodejs/node/pull/53725
+export const SUPPORTS_TYPE_STRIPPING = major > 23 || (major === 23 && minor > 6);
```

Comment:
to make strip types works, you need to change pnp mjs loader and chain other loaders , because (assumption is) types stripes is implemented as loader also. 
https://github.com/yarnpkg/berry/issues/6595#issuecomment-2545840548

Reply:
test with simple type could written to validates this